counsel.el

```diff
@@ -1910,10 +1910,14 @@ minibuffer."
              (enable-recursive-minibuffers t)
              (x (ivy-read "Env: " cands))
              (path (cdr (assoc x cands))))
-        (insert path)
+        (insert (if (file-accessible-directory-p path)
+                    (file-name-as-directory path)
+                  path))
         (ivy--cd-maybe))
     (insert last-input-event)))
 
+(define-key ivy-minibuffer-map "$" 'counsel-read-env)
```

Comment:
This could overwrite user customisations. If `counsel-read-env` belongs in `ivy-minibuffer-map` then the function should be moved to `ivy.el`.